src/pip/_internal/index/package_finder.py

```diff
@@ -726,67 +791,136 @@ def _sort_links(self, links):
                     no_eggs.append(link)
         return no_eggs + eggs
 
-    def _log_skipped_link(self, link, reason):
-        # type: (Link, str) -> None
-        if link not in self._logged_links:
+    def _sort_links(self, links):
+        # type: (Iterable[Link]) -> List[Link]
+        """
+        Returns elements of links in order, non-egg links first, egg links
+        second, while eliminating duplicates
+        """
+        return PackageFinder._sort_links_static(links)
```

Comment:
Do we still need this function?

Reply:
Yes, because this function is called in `evaluate_links_static`.

Reply:
What I meant is, can we just call `_sort_links_static` directly there and get rid of this function? It does not do anything from what I can tell.